### PR TITLE
Courts and HMCTS Tribunals as Organisations

### PR DESCRIPTION
**Pull request by boffbowsh**
Pivotal: https://www.pivotaltracker.com/story/show/92039418

This PR expands the current Organisations resource to display Courts and (HMCTS) Tribunals (aka Judicial Bodies) on GOV.UK, but outside of `/government/organisations`.

Changes in this branch:

- About Us CIP body is rendered as Govspeak for these new resources under a changed heading of "Who we are"
- Added new Role for Judges
- Distinguish between Tribunals that should exist under `/courts-tribunals` and `/government/organisations`. Both types should be of the Organisation Type of `tribunal_ndpb`. However, if the organisation is a child of `hm-courts-and-tribunals-service` they will be shown under `/courts-tribunals`.
- Add auto-linking to the Comments field of Organisation contacts. This is to allow for nicely formatted links to e.g. Court Finder. This uses the `rinku` gem, as the `auto_link` helper has been removed from Rails.
- Disable Jobs and FOI sections for courts and tribunals.
- Added Beta label to the courts and tribunals pages.

Screenshots:

screencapture-whitehall-frontend-dev-gov-uk-courts-tribunals-1429871424208
screencapture-whitehall-frontend-dev-gov-uk-courts-tribunals-bankruptcy-court-1429871404389
screencapture-whitehall-frontend-dev-gov-uk-courts-tribunals-bankruptcy-court-1429871124867

### CODE REVIEW

**Comment by tekin** [comment]
I thought the plan was for there to be a `CourtsController` and separate templates?

**Comment by boffbowsh** [comment]
That was the plan, but it changed. The functional differences between the two templates were tiny in the end, and we decided to adapt what we have instead. The unfinished branch for the separate controller is [here](https://github.com/alphagov/whitehall/compare/courts-and-tribunals-pages) so you can see how fiddly it started to get.

**Comment by boffbowsh** [comment]
Update from @tekin:
> Yeah, if it's too complicated to do it as a separate controller, that’s fine. If I’m honest though, this was the kind of tangle I was worried about when it was done as another organisation, rather than as a separate model. The organisations controller was already quite a mess and this isn’t making it any better.

IMO there's not a huge amount of change in `OrganisationsController`. Tekin doesn't have time to review this pre-election, can someone else review please?

File: app/controllers/organisations_controller.rb [hunk 1]
```diff
@@ -42,6 +50,7 @@ def show
             @traffic_commissioners = traffic_commissioners
             @chief_professional_officers = chief_professional_officers
             @special_representatives = special_representatives
+            @judges = judges
```
Comment (danielroseman) [inline app/controllers/organisations_controller.rb]: since this will only be used for courts, maybe add `if params[:courts_only]`?

Comment (boffbowsh) [inline app/controllers/organisations_controller.rb]: Good point, will do.

File: app/views/organisations/show.html.erb [hunk 2]
```diff
@@ -17,17 +21,23 @@
                 locals: { feature_list: @feature_list,
                           recently_updated: @recently_updated,
                           organisation: @organisation } %>
-      <section id="what-we-do" class="what-we-do">
+      <section id="<%= @organisation.court_or_hmcts_tribunal? ? 'who-we-are' : 'what-we-do'
+        %>" class="what-we-do">
         <div class="content">
-          <h1 class="label"><%= t('organisation.headings.what_we_do') %></h1>
+          <h1 class="label"><%= t("organisation.headings.#{@organisation.court_or_hmcts_tribunal? ? 'who_we_are' : 'what_we_do'}") %></h1>
```
Comment (danielroseman) [inline app/views/organisations/show.html.erb]: This is pretty hard to read. Could it be extracted into a helper?

Comment (boffbowsh) [inline app/views/organisations/show.html.erb]: Done.

**Comment by danielroseman** [comment]
:+1:  looks good.

**Pull request merged**